wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -700,7 +717,10 @@ class KotlinGenerator private constructor(
       .returns(BOOLEAN)
 
     val body = buildCodeBlock {
-      addStatement("if (%N === this) return·true", otherName)
+      if (!mutableTypes) {
+        // We cannot rely on referential equality if the message is mutable.
```

Comment:
Is this true? I don't think this is true.

Reply:
False. True, it's false. True, that's my fault.

Reply:
Clarified my comment. Please tell me if this reads better.

Reply:
Reworded this.

Reply:
I meant the whole conditional can go I think. Mutable or not, `===` should always be a valid shortcut for `equals` for any object.

Reply:
You are right 😆. Will fix that and send another update.